drivers/adc/adc_mcux_lpadc.c

```diff
@@ -455,8 +470,18 @@ static const struct adc_driver_api mcux_lpadc_driver_api = {
 #endif
 };
 
+#define LPADC_REGULATOR_DEPENDENCY(node_id, prop, idx) \
+	DEVICE_DT_GET(DT_PHANDLE_BY_IDX(node_id, prop, idx)),
+
+#define LPADC_REGULATORS_DEFINE(inst)				\
+	static const struct device *mcux_lpadc_req_regulators_##inst[] = {	\
+		COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, required_regulators),	\
+			(DT_INST_FOREACH_PROP_ELEM(inst, required_regulators,	\
+				LPADC_REGULATOR_DEPENDENCY)), ()) NULL};
 
 #define LPADC_MCUX_INIT(n)						\
+	LPADC_REGULATORS_DEFINE(n)						\
```

Comment:
This is a bit of a nitpick, but I would rather see something like the following here:
```
static const struct device *mcux_lpadc_req_regulators_##inst[] = {LPADC_REGULATORS_DEFINE(inst)
```

Personally I find that this makes the macros above easier to follow. Feel free not to take the suggestion though